ksiren/src/main/java/com/brightspace/ksiren/Action.kt

```diff
@@ -15,14 +15,35 @@ package com.brightspace.ksiren
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+interface ActionBase : JsonSerializable {
+	val name: String
+	val classes: List<String>
+	val method: String
+	val href: String
+	val title: String?
+	val type: ContentType
+	val fields: List<FieldBase>
+
+	fun hasField(name: String): Boolean = this.fields.any() { it.name == name }
+
+	fun toJsonRequestBody(): String? =
+		if (fields?.isNotEmpty())
+			fields?.mapNotNull { field -> field.value?.let { json -> field.name to json } }
```

Comment:
Do we need the null checking `?` on `fields`? Is it possible to get past the above `if` statement when `fields` is null?

Reply:
`fields` shouldn't even be able to be null here - it's not nullable in the declaration

Reply:
Yep, I think leftover from refactoring. Surprised the compiler didn't complain here, but maybe it did and I missed it.